lm_engine/modeling_utils/sequence_mixer_blocks/gru/triton_implementation/forward.py

```
@@ -0,0 +1,229 @@
+        assert max_seqlen is None
+        B, S, _, H = x.size()
+    else:
+        B = cu_seqlens.size(0) - 1
+        S = max_seqlen
+        H = x.size(-1)
+
+    Nx, Nxf, Nxr, Nw, Nwf, Nwr, N = _get_num_heads(x=x, W=W, xf=xf, Wf=Wf, xr=xr, Wr=Wr, run_check=False)
+
+    BLOCK_SIZE_H = get_next_power_of_2(H)
+    BLOCK_SIZE_H = max(16, BLOCK_SIZE_H)
+    GRID = lambda kwargs: (ceil_divide(B, kwargs["BLOCK_SIZE_B"]), N)
```

Ruff (0.16.3)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `GRID` as a `def`

(E731)